src/merkle_proof/poseidon_branch.rs

```diff
@@ -32,7 +34,9 @@ where
     H: ByteHash,
 {
     fn from(branch: &Branch<C, H>) -> PoseidonBranch {
-        let mut poseidon_branch = PoseidonBranch::with_capacity(branch.levels().len() - 1);
+        let mut poseidon_branch =
+            PoseidonBranch::with_capacity(POSEIDON_BRANCH_LENGTH);
```

Comment:
As said, this should be provided by the user that calls the from impl or similar. Can't be hardcored since we can have trees with different heights.

Reply:
Maybe we can just generate it with the minimin height that we actually have (Bid tree with height = 14) and we can always re-allocate if needed dinamically.

Reply:
I think this is premature optimization and just using a normal Vec::new() is just fine

Reply:
Makes sense!